### ISSUE TITLE
Backups not syncing to new cluster for migration if owner reference does not exist

### ISSUE THREAD

**Issue by deefdragon** [vmware-tanzu/velero#7031]
**What steps did you take and what happened:**
I created a kubernetes cluster on Digital ocean to verify my migration and backup plans, and when attempting to validate my migration plan, the backups were recognized by velero and logged as imported, but do not show up in the cluster

**What did you expect to happen:**

velero backup describe <backup-name> to show the information for the backup that was logged as successfully synced to the cluster.

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

> An error occurred: backups.velero.io "<backupname>" not found

**Anything else you would like to add:**
The logs specifically show all of the pre-existing backups as successfully migrated ([this log line](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/backup_sync_controller.go#L184)), but none of the backups show up in any of the methods for checking for them. lens, kubectl, velero describe, etc.

**Environment:**

- Velero version: 1.12
- Velero features (use `velero client config get features`): NOT SET
- Kubernetes version (use `kubectl version`): 1.23 on old server, 1.28 on new
- Kubernetes installer & version: kubeadm install to digital ocean
- Cloud provider or hardware configuration: moving from local to digital ocean
- OS (e.g. from `/etc/os-release`): unknown

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by deefdragon** [comment]
I was able to duplicate this with a local cluster & think I have determined the root cause. 

I decided to attempt to add the backup manually to see if I could get some better logs for the exact reason. After some digging, I was able to get the backup to load in via kubectl apply **if I deleted the owner reference**. The backups I use are created from a schedule, and, as the schedule does not exist in the destination cluster, it appears that it would not create.

I think an extra step to remove the owner reference needs to be added to [the code here](https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/backup_sync_controller.go#L142-L175) to check for the owner references all exist, and if they do not, remove them.
